Kubernetes/HOWTO-on-prem.md

```diff
@@ -0,0 +1,103 @@
+
+    ~/kube/certs $ ./generate-certs.sh 10.123.45.67   # example! replace
+
+### Allow Routing ###
+_This step may be optional, depending on whether or not you've configured your intended cluster subnet to be routable already_.
+
+Windows nodes will each grab a /24 from the cluster, so if your cluster CIDR is 192.168.0.0/16, a Windows node will use 192.168.0.1/24. Just run:
+
+    ~/kube $ generate-routes.sh 192.168
+
+### Prepare Manifests & Addons ###
+In the `manifest` folder, run the Python script, passing your master IP and the _full_ cluster CIDR:
+
+    $ python2 generate.py 10.123.45.67 192.168.0.0/16
+
+This will generate a set of YAML files. You should [re]move the Python script so that K8s doesn't mistake it for a manifest and cause problems.
+
+### Configure & Run Kubernetes ###
+Run the script `configure-kubectl.sh`, passing your external IP again, to create a configuration in `~/.kube/config` containing all of the certificate data and other parameters:
+
+    ~/kube $ ./configure-kubectl.sh 10.123.45.67
+
+Run the script `start-kubelet.sh`. You may need to run it with `sudo`. Then, in another shell, copy the Kubernetes configuration file to the local kubelet mount path:
+
+    $ sudo cp ~/.kube/config kubelet/
+
+**TODO**: Why is this necessary? Shouldn't this be done automatically by one of the containers?
+
+In another terminal session, run the Kubeproxy script, passing your cluster CIDR (you may also need to run this one with `sudo`):
+
+    ~/kube $ ./kubeproxy.sh 192.168
+
+After some time, you should see the following system state:
+
+  - Under `docker ps`, you should see worker and pod containers mirroring the configs in `manifest/` and in `addons/`.
+  - A call to `~/kube/bin/kubectl cluster-info` should show info for the Kubernetes master, DNS, and heapster.
+  - A new interface `cbr0` under `ipconfig`, with your chosen cluster CIDR.
+
+## Join a Windows Node ##
+Like for Linux, we need an assortment of scripts to prepare things for us. They can be found [here](https://github.com/Microsoft/SDN/tree/master/Kubernetes/windows). Place these in a new folder, `C:\k\`. We also need to copy the .exe binaries we compiled earlier (which we placed in `~/kube/`), as well as the Kubectl config file (from `~/.kube/config`), into this folder.
+
+### Build Docker Image ###
+We need to build the docker image for the Kubernetes infrastructure. Navigate
+to `C:\k\` and run:
+
+    C:\k> docker pull microsoft/windowsservercore-insider:latest
+    C:\k> docker tag [SHA from previous cmd] microsoft/windowsservercore:latest
+    C:\k> docker build -t kublet/pause .
+
+### Join to Cluster ###
+In two PowerShell windows, simply run:
+
+    PS> ./start-kubelet.ps1 -ClusterCidr [Cluster CIDR]

```

Comment:
Elsewhere in the guide we differentiate between passing full CIDR or just the first two octets...maybe we should be explicit here too